### ISSUE TITLE
Concurrent runs on same host cause issues with the downloaded .zip file

### ISSUE THREAD

**Issue by sbyrnes-weblogix** [giggio/node-chromedriver#239]
Is it possible for the download to /tmp to have a unique name (using something like the detetimestamp)

This would help with attempting to isolate these kinds of errors.

I have seen an issue where 2 jobs run on a single host at the same time and one will download the file, the other will also download the file and while the 2nd instance is streaming to the /tmp the first attempts to unzip and causes the file to appear corrupt as its being downloaded.

Thanks

**Comment by giggio** [comment]
Yes, during the download phase, but not at the end, as the name is used for caching. Something would have to be built around it, because at the end we may still end up on a race condition. But we might be able to isolate it so the problem is less frequent.
